File: back/auth-service/src/main/java/com/glucovion/authservice/service/AuthService.java
package com.glucovion.authservice.service;

import com.glucovion.authservice.dto.AppUserRegisterDto;
import com.glucovion.authservice.dto.AuthResponseDto;
import com.glucovion.authservice.dto.LoginRequestDto;

/**
 * Service interface for handling user authentication and registration.
 * <p>
 * Defines operations to:
 * <ul>
 *   <li>Register a new user</li>
 *   <li>Authenticate an existing user and return a JWT token</li>
 * </ul>
 */
public interface AuthService {

    /**
     * Registers a new user and returns an authentication response with a JWT token.
     *
     * @param registerDto the registration request data
     * @return a response containing the JWT token and user details
     */
    AuthResponseDto register(AppUserRegisterDto registerDto);

    /**
     * Authenticates a user based on login credentials.
     *
     * @param loginRequest the login credentials
     * @return a response containing the JWT token and user details
     */
    AuthResponseDto login(LoginRequestDto loginRequest);
}

File: front/src/app/services/auth.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Login } from "../model/login.model";
import { AuthResponse } from "../model/AuthResponse.model";

/**
 * Service responsible for handling authentication logic in the application.
 * Communicates with the backend API to perform login operations.
 */
@Injectable({ providedIn: 'root' })
export class AuthService {

  /** Base URL for authentication-related endpoints */
  private apiUrl = 'http://localhost:8080/api/auth/'; 
  
  /**
   * Injects Angular's HttpClient to perform HTTP requests.
   * @param http The HttpClient service used for API communication.
   */
  constructor(private http: HttpClient) {}

  /**
   * Sends user credentials to the backend to perform login.
   * @param loginData Object containing user email and password.
   * @returns An Observable emitting an AuthResponse containing token and user details.
   */
  login(loginData: Login): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(`${this.apiUrl}login`, loginData);
  }
}
